lib/new_relic/harvest/telemetry_sdk/dimensional_metrics/harvester.ex

```diff
@@ -0,0 +1,87 @@
+defmodule NewRelic.Harvest.TelemetrySdk.DimensionalMetrics.Harvester do
+  use GenServer
+
+  @moduledoc false
+
+  alias NewRelic.Harvest
+  alias NewRelic.Harvest.TelemetrySdk
+
+  @interval_ms 5_000
+
+  @valid_types [:count, :gauge, :summary]
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [])
+  end
+
+  def init(_) do
+    {:ok,
+     %{
+       start_time_ms: System.system_time(:millisecond),
+       metrics: []
+     }}
+  end
+
+  # API
+
+  @spec report_dimensional_metric(:count | :gauge | :summary, atom() | binary(), any, map()) ::
+          :ok
+  def report_dimensional_metric(type, name, value, attributes) when type in @valid_types do
```

Comment:
The main risk I'm noticing in this PR is that `report_dimensional_metric` collects all the metrics and sends every one up, while `report_custom_metric ` will aggregate each metric during the harvest cycle and only sends each one once.

Which behavior do you want this function to have? I suspect most folks would assume the aggregated behavior. For example, if someone calls this function for every single request, we wouldn't want to copy the name & attributes potentially thousands of times a second..

If you did want that behavior, perhaps `send_dimensional_metric` is more expressive of what will happen?

Reply:
Yeah. I definitely agree that it's not what I would expect.

What are your thoughts on efficiently storing the map of attributes along with the metric name?

Eg I report metric_a with attributes a, b, and c. Then I report metric_a with attributes c, d, and e.

Do I merge the maps? Or does that remain as two metrics?

Reply:
I could use something like phash2 on the attributes map and then aggregate matching ones.

Reply:
I think that's right, the whole of the attributes makes up a unique metric. The other metric harvester has an optimized aggregation implementation you could steal

Reply:
So it would be keyed off of a tuple of the name and the hash of the attributes? The timeslice metric harvester uses the name and the scope, it seems.

Reply:
Yep!